code/modules/integrated_electronics/core/assemblies.dm

```diff
@@ -0,0 +1,372 @@
+	return FALSE
+
+/obj/item/device/electronic_assembly/drone/can_move()
+	return TRUE
+
+/obj/item/device/electronic_assembly/update_icon()
+	if(opened)
+		icon_state = initial(icon_state) + "-open"
+	else
+		icon_state = initial(icon_state)
+/*
+/obj/item/device/electronic_assembly/GetAccess()
+	. = list()
+	for(var/obj/item/integrated_circuit/part in contents)
+		. |= part.GetAccess()
+
+/obj/item/device/electronic_assembly/GetIdCard()
+	. = list()
+	for(var/obj/item/integrated_circuit/part in contents)
+		var/id_card = part.GetIdCard()
+		if(id_card)
+			return id_card
+*/
+/obj/item/device/electronic_assembly/examine(mob/user)
+
+	for(var/obj/item/integrated_circuit/IC in contents)
+		IC.external_examine(user)
+	for(var/obj/item/integrated_circuit/output/screen/S in contents)
+		if(S.stuff_to_display)
+			to_chat(user, "There's a little screen labeled '[S]', which displays '[S.stuff_to_display]'.")
+		if(opened)
+			interact(user)
+
+/obj/item/device/electronic_assembly/proc/get_part_complexity()
+	. = 0
+	for(var/obj/item/integrated_circuit/part in contents)
+		. += part.complexity
+
+/obj/item/device/electronic_assembly/proc/get_part_size()
+	. = 0
+	for(var/obj/item/integrated_circuit/part in contents)
+		. += part.size
+
+// Returns true if the circuit made it inside.
+/obj/item/device/electronic_assembly/proc/add_circuit(var/obj/item/integrated_circuit/IC, var/mob/user)
+	if(!opened)
+		to_chat(user, "<span class='warning'>\The [src] isn't opened, so you can't put anything inside.  Try using a crowbar.</span>")
+		return FALSE
+
+	if(IC.w_class > src.w_class)
+		to_chat(user, "<span class='warning'>\The [IC] is way too big to fit into \the [src].</span>")
+		return FALSE
+
+	var/total_part_size = get_part_size()
+	var/total_complexity = get_part_complexity()
+
+	if((total_part_size + IC.size) > max_components)
+		to_chat(user, "<span class='warning'>You can't seem to add the '[IC]', as there's insufficient space.</span>")
+		return FALSE
+	if((total_complexity + IC.complexity) > max_complexity)
+		to_chat(user, "<span class='warning'>You can't seem to add the '[IC]', since this setup's too complicated for the case.</span>")
+		return FALSE
+
+	if(!user.transferItemToLoc(IC, src))
+		return FALSE
+
+	IC.assembly = src
+
+	return TRUE
+
+/obj/item/device/electronic_assembly/afterattack(atom/target, mob/user, proximity)
+	if(proximity)
+		var/scanned = FALSE
+		for(var/obj/item/integrated_circuit/input/sensor/S in contents)
+			S.set_pin_data(IC_OUTPUT, 1, weakref(target))
+			S.check_then_do_work()
+			if(S.scan(target))
+				scanned = TRUE
+		if(scanned)
+			visible_message("<span class='notice'>\The [user] waves \the [src] around [target].</span>")
+
+/obj/item/device/electronic_assembly/attackby(var/obj/item/I, var/mob/user)
+	if(istype(I, /obj/item/integrated_circuit))
+		if(!user.canUnEquip(I))
+			return FALSE
+		if(add_circuit(I, user))
+			to_chat(user, "<span class='notice'>You slide \the [I] inside \the [src].</span>")
```

Comment:
In general you can drop \the if you already use object in interpolated string since byond already handles that by default.